lib/assets/javascripts/turbograft/component_url.js

```diff
@@ -0,0 +1,56 @@
+/* The ComponentUrl class converts a basic URL string into an object
+ * that behaves similarly to document.location.
+ *
+ * If an instance is created from a relative URL, the current document
+ * is used to fill in the missing attributes (protocol, host, port).
+ */
+window.ComponentUrl = (function() {
```

Comment:
Why is this using an IIFE instead of a `class` defined class like `Click` does?

Reply:
Agree, this seems odd. Classes are also not supported by some browsers, but maybe that's ok: https://caniuse.com/?search=class

Reply:
That is what coffeescript generated. If I use class, tests start to fail, no idea why, so I just kept the code as coffeescript generates.

Reply:
It might be related to function binding, but it hard to guess without output. We could also open an issue for this and do it later.

Reply:
I see no usage of binding in coffeescript, and the code on this class is so simple that I don't see how that would influence the behavior to the point that tests would fail.

Reply:
I think I fixed this in my last commit.